src/amp-react-accordion.js

```diff
@@ -0,0 +1,195 @@
+  AmpContext,
+} from './amp-context.js';
+import {Slot} from './slot.js';
+import {withAmpContext} from './amp-context.js';
+
+const {
+  useContext,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} = preactHooks;
+
+let autokeyCounter = 0;
+
+
+export const AmpAccordionContext = preact.createContext();
+
+
+export function AmpAccordion(props) {
+  const {multiple} = props;
+  const [expanded, setExpanded] = useState({});
+  return preact.createElement(
+    'div',
+    {
+      ...props,
+    },
+    preact.createElement(
+      AmpAccordionContext.Provider,
+      {
+        ...props,
+        value: {
+          multiple,
+          // TBD: controlled values require override of properties.
+          expanded,
+          setExpanded,
+        },
+      }
+    )
+  );
+}
+
+AmpAccordion.Section = function(props) {
+  const autokey = useRef(++autokeyCounter).current;
+  const {expanded: expandedProp, defaultExpanded, onExpand} = props;
+  const accordionContext = useContext(AmpAccordionContext);
+  const {expanded: expandedMap, setExpanded: setExpandedMap, multiple} = accordionContext;
+  const expanded =
+      expandedProp !== undefined ? expandedProp :
+      Object.keys(expandedMap).length > 0 ? !!expandedMap[autokey] :
+      defaultExpanded !== undefined ? defaultExpanded :
+      false;
+  const sectionProps = {
+    ...props,
+    header: undefined,
+    expanded,
+  };
+  const headerProps = {
+    onClick: () => {
+      if (onExpand) {
+        onExpand();
+      }
+      toggleExpanded(autokey, multiple, expandedMap, setExpandedMap);
+    },
+  };
+  return preact.createElement(
+    props.type || props.tagName || 'section',
+    sectionProps,
+    [
+      preact.createElement(
+        'span',
+        headerProps,
+        props.header
+      ),
+      preact.createElement(
+        withAmpContext,
+        {
+          renderable: expanded,
+          playable: expanded,
+        },
+        preact.createElement(
+          'div',
+          {
+            style: {
+              display: expanded ? 'contents' : 'none',
+            },
+          },
+          props.children
+        )
+      ),
+    ]);
+}
+
+function toggleExpanded(autokey, multiple, expandedMap, setExpandedMap) {
+  const current = !!expandedMap[autokey];
+  if (multiple) {
+    setExpandedMap(Object.assign(
+      {},
+      expandedMap,
+      {[autokey]: !current}
+    ));
+  } else {
+    setExpandedMap({[autokey]: !current});
+  }
+}
+
+
+const AmpReactAccordion = ReactCompatibleBaseElement(AmpAccordion, {
+  className: 'i-amphtml-fill-content i-amphtml-replaced-content',
+  attrs: {
+  },
+  template: true,
+  // TODO: simply override ReactBaseElement instead.
+  init(reactBaseElement) {
+    const {element} = reactBaseElement;
+    const mu = new MutationObserver(records => {
+      rebuild();
+    });
+    const rebuild = () => {
+      const children = [];
+      const sectionChildren = element.querySelectorAll(':scope > section');
```

Comment:
Nit: I don't think `:scope` is necessary for this selector.

Reply:
There's no way to do a direct descendant selector without `:scope` in this case. Though in PROD code we may want to switch to just walking over `.children` given the `:scope` is not polyfillable.